src/SwedbankPay.Sdk.Infrastructure/JsonSerialization/Converters/CustomMetaDataConverter.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SwedbankPay.Sdk.JsonSerialization.Converters
+{
+    public class CustomMetaDataConverter : JsonConverter<MetadataResponse>
+    {
+        public override MetadataResponse Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException();
+            }
+
+            var metadata = new MetadataResponse();
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    return metadata;
+                }
+
+                string keyString = reader.GetString();
+                reader.Read();
+
+                if(reader.TokenType == JsonTokenType.String)
+                {
+                    string itemValue = reader.GetString();
+
+                    if (keyString.Equals(nameof(MetadataResponse.Id), StringComparison.InvariantCultureIgnoreCase))
+                        metadata.Id = itemValue;
+
+                    metadata.Add(keyString, itemValue);
+                }
+                else if (reader.TokenType == JsonTokenType.Number)
+                {
+                    if (reader.TryGetInt64(out var valueDouble))
+                    {
+                        metadata.Add(keyString, valueDouble);
```

Comment:
Won't decimal numbers parse without error to integer, thus discarding valuable data? Can't we just always parse to `decimal` to ensure precision and data integrity and leave it to the consumer to convert to integer if they want to?

```suggestion
                    if (reader.TryGetInt64(out var valueInt))
                    {
                        metadata.Add(keyString, valueInt);
```